diff --git a/corelib/mtd-interface.c b/corelib/mtd-interface.c
index c8a9e45..286d757 100644
--- a/corelib/mtd-interface.c
+++ b/corelib/mtd-interface.c
@@ -56,8 +56,20 @@ int get_mtd_from_device(char *s) {
 		return -1;
 
 	ret = sscanf(s, "mtd%d", &mtdnum);
-	if (ret <= 0)
-		ret = sscanf(s, "/dev/mtd%d", &mtdnum);
+    TRACE("s=%s, ret=%d\n", s, ret);
+	if (ret <= 0){
+        char buf[512];
+        memset(buf, 0, sizeof(buf));
+        char *res = realpath(s, buf);
+        if(res){
+            TRACE("buf=%s\n", buf);
+		    ret = sscanf(buf, "/dev/mtd%d", &mtdnum);
+        }
+        else
+        {
+            return -1;
+        }
+        }
 
 	if (ret <= 0)
 		return -1;
